main thread dies after daemon() and pthread_create()
patch doesn't address static linking
see https://bugs.busybox.net/show_bug.cgi?id=193 for details

--- include/unistd.h.orig	2010-04-02 17:34:27.000000000 +0200
+++ include/unistd.h	2011-04-24 20:36:44.804898073 +0200
@@ -758,7 +758,7 @@
    Return -1 for errors, 0 to the new process,
    and the process ID of the new process to the old process.  */
 extern __pid_t fork (void) __THROW;
-libc_hidden_proto(fork)
+//libc_hidden_proto(fork)
 #endif
 
 #if defined __USE_BSD || defined __USE_XOPEN_EXTENDED
--- libc/sysdeps/linux/common/fork.c.orig	2011-04-24 20:38:17.114898074 +0200
+++ libc/sysdeps/linux/common/fork.c	2011-04-24 20:37:43.144898072 +0200
@@ -17,7 +17,7 @@
 #define __NR___libc_fork __NR_fork
 _syscall0(pid_t, __libc_fork)
 weak_alias(__libc_fork,fork)
-libc_hidden_weak(fork)
+//libc_hidden_weak(fork)
 #endif
 
 #elif defined __UCLIBC_HAS_STUBS__
